server/api/app.go

```diff
@@ -18,9 +18,67 @@ func (at AppType) IsValid() bool {
 		at == AppTypeBuiltin
 }
 
+// FunctionType describes different functions that could be executed by the App.
+// For now we support AWS lambda function and an http call.
+type FunctionType string
+
+const (
+	awsLambdaFunction FunctionType = "aws_lambda"
+	httpFunction      FunctionType = "http"
+)
+
+func (ft FunctionType) IsValid() bool {
+	return ft == awsLambdaFunction ||
+		ft == httpFunction
+}
+
+// AssetType describes static assets of the Mattermost App.
+// Assets can be saved in S3 with appropriate permissions,
+// or they could be fetched as ordinary http resources.
+type AssetType string
+
+const (
+	s3Asset   AssetType = "s3_asset"
+	httpAsset AssetType = "http_asset"
+)
+
+func (at AssetType) IsValid() bool {
+	return at == s3Asset ||
+		at == httpAsset
+}
+
+// AppStatus describes status of the app
+type AppStatus string
+
+const (
+	AppStatusRegistered AppStatus = "registered"
```

Comment:
1/5 `s/registered/listed/` or even, `known`?

Reply:
It was named `listed`, Catalin suggested to rename it to `registered`. I'm 0/5 on either one.

Reply:
:) let's leave it as then, at least for the time being.

Reply:
I was 1/5 on it - listed seems to be the already established name. Either way it's fine from my pov.